### PR TITLE
Added System.Collections.IDictionary (non-generic) to ObservableDicti…

### PR DESCRIPTION
**Pull request by russell-elledge**
Added System.Collections.IDictionary (non-generic) to ObservableDictionary
Added DictionaryTests.cs along with test methods for all relevant ObservableDictionary additions related to IDictionary implementation.
Updated UpdateCollectionHelper so that it implements System.Collections.ICollection along with ICollection<TType>.
Project file addition of DictionaryTests.cs.

### CODE REVIEW

File: Assisticant/Collections/Impl/UpdateCollectionHelper.cs [hunk 1]
```diff
@@ -45,26 +46,37 @@ public void CopyTo(T[] array, int arrayIndex)
...
+
+        public bool IsSynchronized => false;
+        public object SyncRoot => this;
+
+        public bool IsReadOnly
 		{
-			get { return true; }
+			get { return false; }
 		}
 		public bool Remove(T item)
```
Comment (russell-elledge) [inline Assisticant/Collections/Impl/UpdateCollectionHelper.cs:67]: Just noticed this.  If you aren't wanting to support modification methods in the UpdateCollectionHelper, then this Remove method should go back to being a throw new NotSupportedException.

Comment (michaellperry) [reply Assisticant/Collections/Impl/UpdateCollectionHelper.cs:67]: The intent here is to treat the Dictionary.Values and Dictionary.Keys collections as read-only. If you want to modify the dictionary, you should do so with the object itself. 

I accepted the PR before thinking this through. I think I will revert to the throw.

Comment (russell-elledge) [reply Assisticant/Collections/Impl/UpdateCollectionHelper.cs:67]: Sounds good.  Thank you

**Comment by michaellperry** [comment]
The new package is published at nuget.org. Thank you!

**Pull request merged**